src/AppInstallerCLICore/CheckpointManager.cpp

```diff
@@ -0,0 +1,150 @@
+        {
+            automaticCheckpoint.Set(AutomaticCheckpointData::Command, {}, std::string{ executingCommand->FullName() });
+        }
+
+        const auto& argTypes = context.Args.GetTypes();
+        for (auto type : argTypes)
+        {
+            const auto& argument = std::to_string(static_cast<int>(type));
+            auto argumentType = Argument::ForType(type).Type();
+
+            if (argumentType == ArgumentType::Flag)
+            {
+                automaticCheckpoint.Set(AutomaticCheckpointData::Arguments, argument, {});
+            }
+            else
+            {
+                const auto& values = *context.Args.GetArgs(type);
+                automaticCheckpoint.SetMany(AutomaticCheckpointData::Arguments, argument, values);
+            }
+        }
+    }
+
+    std::optional<Checkpoint<AutomaticCheckpointData>> CheckpointManager::GetAutomaticCheckpoint()
+    {
+        const auto& checkpointIds = m_checkpointDatabase->GetCheckpointIds();
+        if (checkpointIds.empty())
+        {
+            return {};
+        }
+
+        CheckpointDatabase::IdType automaticCheckpointId = checkpointIds.back();
+        return Checkpoint<AutomaticCheckpointData>{ m_checkpointDatabase, automaticCheckpointId };
+    }
+
+    Checkpoint<CLI::Execution::Data> CheckpointManager::CreateCheckpoint(std::string_view checkpointName)
+    {
+        CheckpointDatabase::IdType checkpointId = m_checkpointDatabase->AddCheckpoint(checkpointName);
+        Checkpoint<CLI::Execution::Data> checkpoint{ m_checkpointDatabase, checkpointId };
+        return checkpoint;
+    }
+
+    std::vector<Checkpoint<CLI::Execution::Data>> CheckpointManager::GetCheckpoints()
+    {
+        auto checkpointIds = m_checkpointDatabase->GetCheckpointIds();
+
+        // Remove the last checkpoint (automatic)
+        checkpointIds.pop_back();
+
+        std::vector<Checkpoint<CLI::Execution::Data>> checkpoints;
+        for (const auto& checkpointId : checkpointIds)
+        {
+            checkpoints.emplace_back(Checkpoint<CLI::Execution::Data>{ m_checkpointDatabase, checkpointId });
+        }
+
+        return checkpoints;
+    }
+
+    void CheckpointManager::CleanUpDatabase()
+    {
+        if (m_checkpointDatabase)
+        {
+            m_checkpointDatabase.reset();
+        }
+
+        if (!m_resumeId.empty())
+        {
+            const auto& checkpointDatabasePath = GetCheckpointDatabasePath(m_resumeId);
+
+            if (std::filesystem::exists(checkpointDatabasePath))
+            {
+                std::error_code error;
+                if (std::filesystem::remove(checkpointDatabasePath, error))
+                {
+                    AICLI_LOG(CLI, Info, << "Checkpoint database deleted: " << checkpointDatabasePath);
+                }
+
+                const auto& checkpointDatabaseParentDirectory = checkpointDatabasePath.parent_path();
+                if (std::filesystem::remove(checkpointDatabaseParentDirectory, error))
```

Comment:
I feel like I've left this comment before, but why not simply delete the directory with a recursive delete method?